### ISSUE TITLE
Engine Question

### ISSUE THREAD

**Issue by eliddell1** [Ark444/YateBTS_smssend#5]
hello,
hope all is well.  I am working on that feature i requested, and was wondering if you know How to get the list of registered MSDINs using the Engine?

Trouble finding Documentation on the engine.   Certainly it should be possible to get the list and save it as a var from your module no?

thanks in advance
eliddell

**Comment by eliddell1** [comment]
I can read the file but that seems cumbersome

**Comment by trpyn** [comment]
Hey,

I think you can probably get these using a Message ( http://docs.yate.ro/wiki/Messages ), the list of messages being here http://docs.yate.ro/wiki/Standard_Messages
However I didn't see one that's explicitly says "msisdn_listing". 

I'll take a look at the plugin that generates the list when using the command `nib list register`

**Comment by eliddell1** [comment]
Yeah doesn't look like there is such a message.  Where is the nib code? I saw the nib.js opens a .conf files and parses it.. don't know enough about c to do that cleanly.  
Wish one could just call nib list registered and parses the output.  But the return is just a Boolean

**Comment by trpyn** [comment]
I did not have much time to look into it, and won't be available for a bit less than a week, so I did not find where the nib list registered function is implemented in the code.
You can still get the output from the telnet command line: connect to: localhost on port 5038 when running YateBTS and type "nib list registered", that will print the output on the terminal, which you could pipe into a script to get the list.
Sorry I can't help you more for now

**Comment by eliddell1** [comment]
i suppose.. though that would be a mess to parse, and i would rather just be able to type "smssend all <caller> message"

anyway, made a fork with a pull request, that allows for comma separated recipient support for now..
example: smssend 1111111,2222222 411 hello

**Comment by trpyn** [comment]
Hi, 
I'm gonna check how to find the list of MSISDNs directly using yateBTS, we will need it anyway when implementing the `all` thingy. However I do believe it would not be much more work to handle regex in the MSISDN field.
I've reviewed your pull request, sorry I don't want to merge it as is.
